File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/template/MainTemplateTest.java
package designPatterns.behavioral.template;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainTemplateTest {
  private Super sup;

  @Test
  void equalsTest() {
    sup = new ConcreteSub.A();
    assertThat(sup.play()).isEqualTo("A initialized!\nA started.\nA finished!");

    sup = new ConcreteSub.B();
    assertThat(sup.play()).isEqualTo("B initialized!\nB started.\nB finished!");
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/template/MainTemplateTest.kt
package behavioral.template

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MainTemplateTest {
  private lateinit var sup: Super

  @Test
  internal fun equalsTest() {
    sup = ConcreteSub.A()
    assertThat(sup.play()).isEqualTo("A initialized!\nA started.\nA finished!")

    sup = ConcreteSub.B()
    assertThat(sup.play()).isEqualTo("B initialized!\nB started.\nB finished!")
  }
}